server/db/models/user.js

```diff
@@ -49,14 +54,18 @@ var encryptPassword = function (plainText, salt) {
 };
 
 schema.pre('save', function (next) {
-
     if (this.isModified('password')) {
         this.salt = this.constructor.generateSalt();
         this.password = this.constructor.encryptPassword(this.password, this.salt);
     }
-
+    this.constructor.find({email: this.email}, function(err, user){
```

Comment:
@johnhillphoto .find() returns an array. If it finds no accounts associated to the e-mail, will it return an array of length 0? or will it return undefined? I think findOne might make more sense here because it will either be the entry or it will be null/undefined, right?